[create-sitecore-jss] Cleanup changes for scss SXA files

MenKNas opened the pull request:
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/content-sdk/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation
This PR focuses on streamlining the SCSS structure by removing dead code, improving performance, and simplifying the file organization. Key improvements include:

  - Removed unused `variants/` directory and all empty or redundant `index.scss` files
  - Consolidated font imports: kept Roboto, removed Open Sans to reduce bundle size (since not both are used)
  - Cleaned up unused imports across stylesheets
  - Improved file structure and import consistency for better maintainability

## Testing Details
- [ ] Unit Test Added
- [ ] Manual Test/Other (Please elaborate)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

MenKNas committed:
[create-sitecore-jss] cleaup changes for scss SXA files

MenKNas committed:
[create-sitecore-jss] update CHANGELOG.md

illiakovalenko reviewed (changes requested):
1) [app.css](https://github.com/Sitecore/content-sdk/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/app.css) can be removed, since it's not referenced in the app. It has Experience Editor and styleguide related styles
2) Are fonts imported in [_basic/main.scss_](https://github.com/Sitecore/content-sdk/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/basic/main.scss#L1)? Is it the same import? Can extra import be removed [here](https://github.com/Sitecore/content-sdk/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/main.scss#L1)
3) Let's remove this extra comment: https://github.com/Sitecore/content-sdk/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/basic/_promo.scss#L3
4) Should these editing related styles be removed? I don't think it's used in Pages: https://github.com/Sitecore/content-sdk/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/basic/_component.scss#L34
5) One of our goals was trying to improve the structure of our styles and make it easy to analyze. We have some component related styles under _basic/*_ and also under _sass/*_.
For example we have a bunch of Navigation component styles (check the rest of components as well):
* https://github.com/Sitecore/content-sdk/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/basic/_navigation.scss
* https://github.com/Sitecore/content-sdk/tree/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/sass/components/navigation
* https://github.com/Sitecore/content-sdk/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/sass/components/_component-navigation.scss
Can we think more about ways to improve the structure of our styles? What are the differences between these styles? Can something be consolidated under one folder or file
Additionally, it would be nice to briefly explain [here](https://github.com/Sitecore/content-sdk/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/main.scss#L1) the purpose and role of each section in the file?
6) Can this container editing related style be removed? Looks like it's applied only for Experience Editor: https://github.com/Sitecore/content-sdk/blob/dev/packages/create-sitecore-jss/src/templates/nextjs/src/assets/sass/components/container/_bordered.scss#L19

MenKNas committed:
[create-sitecore-jss] address PR comments

MenKNas commented:
hey @illiakovalenko, thanks for the feedback. You can see my response to your comments below: 

1) Correct. Removed.

2) Correct. Removed.

3) Removed.

4) Correct. Removed.

5) It looks like those styling files do not fully overlap in what they do, and they do slightly different things. 
For example, everything inside the `basic`  folder seems to be overriding the starter template, everything in `components` provides the core layout and responsiveness, and there are also other wrapper classes in `_component-navigation.scss`. After discussing with Ruslan, we agreed to keep them in their current scopes to avoid any unintended regressions. I’ve also added a comment at the top of `main.scss` explaining each section’s purpose for clarity.

6) Correct. Removed.

MenKNas committed:
[create-sitecore-jss] rename scss folder to globals

illiakovalenko reviewed (approved):
Looks good!
